PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/LoadJsonConfigFile.java

```diff
@@ -48,16 +64,118 @@ private void loadJsonObject() throws IOException {
+                    + ". Please check the file format. Error: " + e.getMessage(), e);
+        } catch (IOException e) {
+            LOGGER.log(Level.SEVERE, "IO error while reading configuration file from path: " + CONFIG_FILE_PATH
+                    + ". Please check file permissions and retry. Error: " + e.getMessage(), e);
         } catch (Exception e) {
-            // Log the error message
-            System.err.println("Error loading JSON configuration: " + e.getMessage());
-            // Rethrow the exception
-            throw e;
+            LOGGER.log(Level.SEVERE, "Unexpected error while loading configuration. " + "Please investigate further. Error: " + e.getMessage(), e);
         }
+
     }
 
-    // The getJsonObject method returns the loaded JSON object
+    /**
+     * The getJsonObject method returns the loaded JSON object
+     * @return JSONObject loaded JSON object
+     */
     public JSONObject getJsonObject() {
         return jsonObject;
     }
+
+
+    /** Method to get the value for a given rule name and key as an Object
+     * If the rule name exists and has the key, return the value
+     * @param ruleName String rule name
+     * @param key String key
+     * @return Object value for the key
+     */
+    private Object getConfigValue(String ruleName, String key) {
+        if (jsonObject.has(ruleName)) {
+            JSONObject ruleConfig = jsonObject.getJSONObject(ruleName);
+            if (ruleConfig.has(key)) {
+                return ruleConfig.get(key);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Method to get the methods to check for a given rule name, which is a List of Strings
+     * @param ruleName String rule name
+     * @return List<String> methods to check
+     */
+    public List<String> getMethodsToCheck(String ruleName) {
+        String key = "methods_to_check";
+        Object value = getConfigValue(ruleName, key);
+        return processMethodsToCheck(value);
+    }
+
+    /** Method to get the method to check for a given rule name, which is a String
+     * If the rule name exists and has the key, return the value
+     * @param ruleName String rule name
+     * @return String method to check
+     */
+    public String getMethodToCheck(String ruleName) {
+        String key = "method_to_check";
+        return (String) getConfigValue(ruleName, key);
+    }
+
+    /**
+     * Method to get the client name for a given rule name
+     * @param ruleName String rule name
+     * @return String client name
+     */
+    public String getClientName(String ruleName) {
+        String key = "client_name";
+        return (String) getConfigValue(ruleName, key);
+    }
+
+    /** Method to get the suggestion message for a given rule name
+     * @param ruleName String rule name
+     * @return String suggestion message
+     */
+    public String getSuggestionMessage(String ruleName) {
+        String key = "antipattern_message";
```

Comment:
Can this be a constant?

Reply:
I'm only using it within this method so I opted to keep it as a local variable

Reply:
Then it can just be inline?